dask_match/tests/test_collection.py

```diff
@@ -0,0 +1,162 @@
+import pandas as pd
+import pytest
+from dask.dataframe.utils import assert_eq
+from dask.utils import M
+
+from dask_match import from_pandas, optimize
+
+
+@pytest.fixture
+def df():
+    df = pd.DataFrame({"x": range(100)})
+    df["y"] = df.x * 10.0
+    yield df
+
+
+@pytest.fixture
+def ddf(df):
+    yield from_pandas(df, npartitions=10)
+
+
+def test_del():
+    df = pd.DataFrame({"x": range(100), "y": range(100)})
+    ddf = from_pandas(df, npartitions=10)
+    df = df.copy()
+
+    # Check __delitem__
+    del df["x"]
+    del ddf["x"]
+    assert_eq(df, ddf)
+
+
+def test_setitem():
+    df = pd.DataFrame({"x": range(100), "y": range(100)})
+    ddf = from_pandas(df, npartitions=10)
+    df = df.copy()
+
+    ddf["z"] = ddf.x + ddf.y
+
+    assert "z" in ddf.columns
+    assert_eq(ddf, ddf)
+
+
+def test_meta_divisions_name():
+    a = pd.DataFrame({"x": [1, 2, 3, 4], "y": [1.0, 2.0, 3.0, 4.0]})
+    df = 2 * from_pandas(a, npartitions=2)
+    assert list(df.columns) == list(a.columns)
+    assert df.npartitions == 2
+
+    assert df.x.sum()._meta == 0
+    assert df.x.sum().npartitions == 1
+
+    assert "mul" in df._name
+    assert "sum" in df.sum()._name
+
+
+def test_meta_blockwise():
+    a = pd.DataFrame({"x": [1, 2, 3, 4], "y": [1.0, 2.0, 3.0, 4.0]})
+    b = pd.DataFrame({"z": [1, 2, 3, 4], "y": [1.0, 2.0, 3.0, 4.0]})
+
+    aa = from_pandas(a, npartitions=2)
+    bb = from_pandas(b, npartitions=2)
+
+    cc = 2 * aa - 3 * bb
+    assert set(cc.columns) == {"x", "y", "z"}
+
+
+def test_dask(df, ddf):
+    assert (ddf.x + ddf.y).npartitions == 10
+    z = (ddf.x + ddf.y).sum()
+
+    assert assert_eq(z, (df.x + df.y).sum())
```

Comment:
We cant use `assert_eq` on `Scalar` objects using `dask:main`. We can do `assert z == (df.x + df.y).sum()` here, modify `dask/dataframe/utils/assert_eq`, or implement our own version of `assert_eq` in dask-match.

Reply:
Ah indeed.  I had that change locally.  

Well, rather than work around this, lets fix the problem: https://github.com/dask/dask/pull/10125

In general it seems like we're pushing a bit more in this direction.  It's probably time to make this effort a bit cleaner in general, and avoid workarounds where we can.